resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/registry/SegmentNode.java

```diff
@@ -446,14 +446,22 @@ else if (!consumeMatch)
       }
       Collections.sort(sortList);
       SortEntry sortEntry = sortList.get(0);
-      if (targetMethods != null && targetMethods.size() > 1)
+      if (targetMethods != null && targetMethods.size() > 1 && !checkExpressionsUnique(sortList))
       {
          LogMessages.LOGGER.multipleMethodsMatch(requestToString(request), methodNames(targetMethods));
       }
       request.setAttribute(RESTEASY_CHOSEN_ACCEPT, sortEntry.getAcceptType());
       return sortEntry.match;
    }
 
+   private boolean checkExpressionsUnique(List<SortEntry> sortList){
+      List<String> list = new ArrayList<>();
+      for(SortEntry elm: sortList){
+         list.add(elm.match.expression.getRegex());
+      }
+      return new HashSet<>(list).size() == list.size();

```

Comment:
@treblereel , do we really need the 'list' instance here? Can't we build up the HashSet directly and compare its size with sortList size? Or even better, at that point can't we simply return the first time that the HashSet#add method invocation returns false?